superset/migrations/versions/96e99fb176a0_add_import_mixing_to_saved_query.py

```diff
@@ -0,0 +1,134 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""add_import_mixing_to_saved_query
+
+Revision ID: 96e99fb176a0
+Revises: af30ca79208f
+Create Date: 2020-10-21 21:09:55.945956
+
+"""
+
+import os
+import time
+from uuid import uuid4
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects.mysql.base import MySQLDialect
+from sqlalchemy.dialects.postgresql.base import PGDialect
+from sqlalchemy.exc import OperationalError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy_utils import UUIDType
+
+from superset import db
+
+# revision identifiers, used by Alembic.
+revision = "96e99fb176a0"
+down_revision = "af30ca79208f"
+
+
+Base = declarative_base()
+
+
+class ImportMixin:
+    id = sa.Column(sa.Integer, primary_key=True)
+    uuid = sa.Column(UUIDType(binary=True), primary_key=False, default=uuid4)
+
+
+class SavedQuery(Base, ImportMixin):
+    __tablename__ = "saved_query"
+
+
+default_batch_size = int(os.environ.get("BATCH_SIZE", 200))
+
+# Add uuids directly using built-in SQL uuid function
+add_uuids_by_dialect = {
+    MySQLDialect: """UPDATE saved_query SET uuid = UNHEX(REPLACE(uuid(), "-", ""));""",
+    PGDialect: """UPDATE saved_query SET uuid = uuid_in(md5(random()::text || clock_timestamp()::text)::cstring);""",
+}
+
+
+def add_uuids(session, batch_size=default_batch_size):
```

Comment:
I was on the fence about doing this, because it requires small changes in the previous script, and changing a previous migration script seems dangerous.

In this case it seems safe enough... let me know what you think.

Reply:
Makes sense. Didn't know updates were needed, in that case I'm OK with either approach.